src/metabase/integrations/slack.clj

```diff
@@ -53,29 +45,18 @@
   "Calls Slack api `users.list` function and returns the list of available users."
   (comp :members (partial GET :users.list)))
 
-(defn- create-files-channel!
-  "Convenience function for creating our Metabase files channel to store file uploads."
-  []
-  (when-let [{files-channel :channel, :as response} (create-channel! files-channel-name)]
-    (when-not files-channel
-      (log/error (u/pprint-to-str 'red response))
-      (throw (ex-info "Error creating Slack channel for Metabase file uploads" response)))
-    ;; Right after creating our files channel, archive it. This is because we don't need users to see it.
-    (u/prog1 files-channel
-      (archive-channel! (:id <>)))))
-
-(defn- files-channel
-  "Return the `metabase_files` channel (as a map) if it exists."
-  []
-  (some (fn [channel] (when (= (:name channel) files-channel-name)
-                        channel))
-        (channels-list :exclude_archived 0)))
-
-(defn get-or-create-files-channel!
-  "Calls Slack api `channels.info` and `channels.create` function as needed to ensure that a #metabase_files channel exists."
+(defn files-channel
+  "Calls Slack api `channels.info` to check whether a channel named #metabase_files exists. If it doesn't,
+   throws an error that advices an admin to create it."
   []
-  (or (files-channel)
-      (create-files-channel!)))
+  (or (some (fn [channel] (when (= (:name channel) files-channel-name)
+                            channel))
+            (channels-list :exclude_archived 0))
+      (let [channel-missing-msg (str "Slack channel named `metabase_files` is missing! Please create the channel in order to proceed with "
+                                     "the Slack integration. The channel is used for storing graphs that are included in pulses and "
+                                     "Metabot anwers.")]
+        (log/error (u/format-color 'red channel-missing-msg))
+        (throw (Exception. channel-missing-msg)))))
```

Comment:
I think you might want to throw an `ex-info` here instead so you can specify the status code that gets returned

```clojure
(throw (ex-info channel-missing-message {:status-code 400}))
```